Source/Scene/Model.js

```diff
@@ -3293,6 +3305,38 @@ define([
         };
     }
 
+    function createClippingPlanesLengthFunction(model) {
+        return function() {
+            return model.clippingPlanes.length;
+        };
+    }
+
+    var scratchCartesian = new Cartesian3();
+    var scratchPlane = new Cartesian4();
+    var scratchMatrix = new Matrix4();
+
+    function createClippingPlanesFunction(model, context) {
+        return function() {
+            var scale = getScale(model, context.uniformState.frameState);
+            Matrix4.multiplyByUniformScale(model.modelMatrix, scale, scratchMatrix);
+            Matrix4.multiply(context.uniformState.view3D, scratchMatrix, scratchMatrix);
+
+            var planes = model.clippingPlanes;
+            var length = planes.length;
+            var packedPlanes = new Array(length);
+            for (var i = 0; i < length; ++i) {
+                var plane = planes[i];
+                Matrix4.multiplyByPointAsVector(scratchMatrix, plane.normal, scratchPlane);
+                Cartesian3.normalize(scratchPlane, scratchPlane);
+                Cartesian3.multiplyByScalar(plane.normal, plane.distance, scratchCartesian);
+                Matrix4.multiplyByPoint(scratchMatrix, scratchCartesian, scratchCartesian);
+                scratchPlane.w = Cartesian3.dot(scratchPlane, scratchCartesian);
+                packedPlanes[i] = scratchPlane.clone();
```

Comment:
One last thing - instead of allocating the `packedPlanes` array for each uniform call, store it as a property of Model and create the Cartesian4's in advance, and then clone directly into those. There may need to be some logic in `update` to handle when the array length changes.